### ISSUE TITLE
Warnings when compiling "scimath.interpolate._interpolate"

### ISSUE THREAD

**Issue by rahulporuri** [enthought/scimath#131]
The following compiler warnings are generated when building `scimath.interpolate._interpolate` (seen when we run `python -m etstool install`)

```
creating build\temp.win-amd64-3.6\Release\scimath\interpolate
C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.28.29333\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -IC:\Users\rporuri\.edm\envs\scimath-test-3.6\lib\site-packages\numpy\core\include -Iscimath/interpolate -IC:\Users\rporuri\.edm\envs\scimath-test-3.6\include -IC:\Users\rporuri\.edm\envs\scimath-test-3.6\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.28.29333\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\cppwinrt" /EHsc /Tpscimath/interpolate/_interpolate.cpp /Fobuild\temp.win-amd64-3.6\Release\scimath/interpolate/_interpolate.obj
_interpolate.cpp
scimath/interpolate/_interpolate.cpp(54): warning C4244: 'argument': conversion from 'npy_intp' to 'int', possible loss of data
scimath/interpolate/_interpolate.cpp(51): warning C4244: 'argument': conversion from 'npy_intp' to 'int', possible loss of data
scimath/interpolate/_interpolate.cpp(113): warning C4244: 'argument': conversion from 'npy_intp' to 'int', possible loss of data
scimath/interpolate/_interpolate.cpp(110): warning C4244: 'argument': conversion from 'npy_intp' to 'int', possible loss of data
scimath/interpolate/_interpolate.cpp(173): warning C4244: 'argument': conversion from 'npy_intp' to 'int', possible loss of data
scimath/interpolate/_interpolate.cpp(170): warning C4244: 'argument': conversion from 'npy_intp' to 'int', possible loss of data
scimath/interpolate/_interpolate.cpp(232): warning C4244: 'argument': conversion from 'npy_intp' to 'int', possible loss of data
scimath/interpolate/_interpolate.cpp(229): warning C4244: 'argument': conversion from 'npy_intp' to 'int', possible loss of data
C:\Users\rporuri\work\github\ets\scimath\scimath\interpolate\interpolate.h(21): warning C4244: '=': conversion from '__int64' to 'int', possible loss of data
scimath/interpolate/_interpolate.cpp(54): note: see reference to function template instantiation 'void linear<double>(T *,T *,int,T *,T *,int)' being compiled
        with
        [
            T=double
        ]
C:\Users\rporuri\work\github\ets\scimath\scimath\interpolate\interpolate.h(57): warning C4244: '=': conversion from '__int64' to 'int', possible loss of data
scimath/interpolate/_interpolate.cpp(113): note: see reference to function template instantiation 'void loginterp<double>(T *,T *,int,T *,T *,int)' being compiled
        with
        [
            T=double
        ]
C:\Users\rporuri\work\github\ets\scimath\scimath\interpolate\interpolate.h(147): warning C4244: 'initializing': conversion from '__int64' to 'int', possible loss of data
scimath/interpolate/_interpolate.cpp(173): note: see reference to function template instantiation 'int window_average<double>(T *,T *,int,T *,T *,int,T)' being compiled
        with
        [
            T=double
        ]
C:\Users\rporuri\work\github\ets\scimath\scimath\interpolate\interpolate.h(155): warning C4244: 'initializing': conversion from '__int64' to 'int', possible loss of data
C:\Users\rporuri\work\github\ets\scimath\scimath\interpolate\interpolate.h(103): warning C4244: 'initializing': conversion from '__int64' to 'int', possible loss of data
scimath/interpolate/_interpolate.cpp(232): note: see reference to function template instantiation 'int block_average_above<double>(T *,T *,int,T *,T *,int)' being compiled
        with
        [
            T=double
        ]
C:\Users\rporuri\work\github\ets\scimath\scimath\interpolate\interpolate.h(113): warning C4101: 'next_x': unreferenced local variable
C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.28.29333\bin\HostX86\x64\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:C:\Users\rporuri\.edm\envs\scimath-test-3.6\libs /LIBPATH:C:\Users\rporuri\.edm\envs\scimath-test-3.6\PCbuild\amd64 "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.28.29333\lib\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.18362.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.18362.0\um\x64" /EXPORT:PyInit__interpolate build\temp.win-amd64-3.6\Release\scimath/interpolate/_interpolate.obj /OUT:build\lib.win-amd64-3.6\scimath\interpolate\_interpolate.cp36-win_amd64.pyd /IMPLIB:build\temp.win-amd64-3.6\Release\scimath/interpolate\_interpolate.cp36-win_amd64.lib
   Creating library build\temp.win-amd64-3.6\Release\scimath/interpolate\_interpolate.cp36-win_amd64.lib and object build\temp.win-amd64-3.6\Release\scimath/interpolate\_interpolate.cp36-win_amd64.exp
Generating code
```

**Comment by mdickinson** [comment]
Yep, those should definitely be fixed - at least the 'npy_intp' to 'int' conversions, anyway; I haven't looked at the others. The chances are that interpolation is currently broken for arrays with more than `2^31` elements, though writing a test for that would be awkward.

**Comment by mdickinson** [comment]
> warning C4101: 'next_x': unreferenced local variable

This one should already have been fixed in #114.
